slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -359,10 +361,12 @@ void SLAMServiceImpl::RunSLAM() {
         bool load_frozen_trajectory =
             action_mode == SLAMServiceActionMode::LOCALIZING;
         {
+            optimizing = (optimize_on_start == true);
             // Load apriori map
             std::lock_guard<std::mutex> lk(map_builder_mutex);
             map_builder.LoadMapFromFile(
                 latest_map_filename, load_frozen_trajectory, optimize_on_start);
+            optimizing = false;
```

Comment:
We need to make sure `optimizing` gets set to `false` even if the above code throws, so the above code should be wrapped in a try/catch.

Reply:
I don't understand. The way I think about this is: If the above code throws, I would assume that cartographer will error out and will not be executed, which is what we want. In that case, we don't care what happens after `map_builder.LoadMapFromFile(...)`.

Am I getting this wrong? How do you think about this?

Reply:
Using a mutex instead that is released when the lock goes out of scope.